lib/Jit/LLILCJit.cpp

```diff
@@ -394,3 +407,100 @@ void __cdecl LLILCJit::fatal(int Errnum, ...) {
 void LLILCJit::signalHandler(void *Cookie) {
   // do nothing
 }
+
+void ObjectLoadListener::getDebugInfoForObject(
+    const ObjectFile &Obj, const RuntimeDyld::LoadedObjectInfo &L) {
+  OwningBinary<ObjectFile> DebugObjOwner = L.getObjectForDebug(Obj);
+  const ObjectFile &DebugObj = *DebugObjOwner.getBinary();
+
+  // Get the address of the object image for use as a unique identifier
+  const void *ObjData = DebugObj.getData().data();
+  DIContext *DwarfContext = new DWARFContextInMemory(DebugObj);
+
+  // Use symbol info to iterate functions in the object.
+  // TODO: This may have to change when we have funclets
+
+  for (symbol_iterator SI = DebugObj.symbol_begin(), E = DebugObj.symbol_end();
+       SI != E; ++SI) {
+    SymbolRef::Type SymType;
+    if (SI->getType(SymType))
+      continue;
+    if (SymType == SymbolRef::ST_Function) {
```

Comment:
Nit: maybe change this block to an early-continue style for consistency with the other checks (i.e. instead of `if (SymType == SymbolRef::ST_Function) { ... }`, make this `if (SymType != SymbolRef::ST_Function) continue;`)